features/ui/button/button.tsx

```diff
@@ -1,9 +1,59 @@
-import { ButtonHTMLAttributes } from "react";
+// Button.tsx
+
+import React, { ButtonHTMLAttributes } from "react";
 import classNames from "classnames";
 import styles from "./button.module.scss";
 
-export function Button(props: ButtonHTMLAttributes<HTMLButtonElement>) {
+export enum ButtonSize {
+  sm = "sm",
+  md = "md",
+  lg = "lg",
+  xlg = "xlg",
+}
+
+export enum ButtonColor {
+  primary = "primary",
+  secondary = "secondary",
+  gray = "gray",
+  empty = "empty",
+  emptygray = "emptygray",
+  error = "error",
+  emptyerror = "emptyerror",
+}
+
+export enum ButtonIcon {
+  none = "none",
+  only = "only",
+  trailing = "trailing",
+  leading = "leading",
+}
+
+type ButtonProps = {
+  children: React.ReactNode;
+  size?: ButtonSize;
+  color?: ButtonColor;
+  icon?: ButtonIcon;
+} & ButtonHTMLAttributes<HTMLButtonElement>;
+
+export function Button({
+  children,
+  size = ButtonSize.md,
+  color = ButtonColor.primary,
+  icon = ButtonIcon.none,
+}: ButtonProps) {
   return (
```

Comment:
Two things I think may be contributing to your issues:

1. You're not spreading any passed `props` to your button, causing your button to not receive the `onClick` event handler that the menu and support buttons are passing to it.
    ```js
    export function Button({
      children,
      size = ButtonSize.md,
      color = ButtonColor.primary,
      icon = ButtonIcon.none,
      ...props // Gets the rest of all props
   }: ButtonProps) {...}
    ```
    ```js
      <div
      {...props}
      className={...}
    >
      {children}
    </div>
2. You really want to return a `<button>` element here instead of a `<div>` for semantics. It'll work as a div, but buttons have special properties when it comes to accessibility and keyboard controls. As the component is, i don't think you could tab to it, although you could enable it by adding the html attribute `tabindex`